owncloud/owncloud.py

```diff
@@ -651,6 +667,111 @@ def get_shares(self, path='', **kwargs):
                 return shares
         raise ResponseError(res)
 
+    def create_user_ajax(self, adminuser, adminpass, user_name, initial_password):
+        """Create a new user with an initial password
+	   This does not use the session context, to prevent "Token expired. Please reload page." errors.
+	   We authenticate here ourselves. In this case no cookie or token is needed.
+	   POST /index.php/settings/ajax/createuser.php --data 'username=user2&password=owncloud&groups=' 
+	"""
+        res = self.__make_single_request(
+                'POST',
+                'index.php/settings/ajax/createuser.php',
+		data = { 'username': user_name, 'password': initial_password, 'groups':'' },
+		auth = { 'username': adminuser, 'password': adminpass }
+	        )
+	# {"data":{"message":"The username is already being used"},"status":"error"}
+	# {"data":{"homeExists":false,"username":"user2","groups":[],"storageLocation":"\/srv\/www\/htdocs\/owncloud\/data\/\/user2"},"status":"success"}
+        if res.status_code == 200:
+	  return res.text
+        raise ResponseError(res)
+
+    def create_user(self, user_name, initial_password):
+        """Create a new user with an initial password via provisioning API.
+	   It is not an error, if the user already existed before.
+	   If you get back an error 999, then the provisioning API is not enabled.
+	"""
+        res = self.__make_ocs_request(
+                'POST',
+                self.OCS_SERVICE_CLOUD,
+                'users',
+                data = { 'password': initial_password, 'userid': user_name }
+                )
+
+	# We get 200 when the user was just created.
+        if res.status_code == 200:
+	    # We get an inner 102 although we have an outer 200 when the user already exists.
+            tree = ET.fromstring(res.text)
+            self.__check_ocs_status(tree, [100, 102])
+	    # return ET.tostring(tree)
+	    return "user '" + user_name + "' is alive"
+        raise ResponseError(res)
+
+    def delete_user(self, user_name):
+        """Deletes a user via provisioning API.
+	   If you get back an error 999, then the provisioning API is not enabled.
+	"""
+        res = self.__make_ocs_request(
+                'DELETE',
+                self.OCS_SERVICE_CLOUD,
+                'users/'+user_name
+                )
+
+	# We get 200 when the user was deleted.
+        if res.status_code == 200:
+	    return "user '" + user_name + "' was deleted"
+
+        raise ResponseError(res)
+
+    def check_user(self, user_name):
```

Comment:
Yes, then make it "user_exists" (use underscores in Python instead of camel case)

The thing I was thinking about is that if the "users?search=user_name" call already returns actual useful user information, it would be nice to return that instead.

Whoever tried to check if the user exists simply needs to check whether it returned None or an actual user.
(or provide a separate function "user_exists" that internally called "get_user" to check it, see here https://github.com/owncloud/pyocclient/blob/master/owncloud/owncloud.py#L588 how "is_share" is calling "get_shares")